core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -108,4 +132,89 @@ static void validateMethodNames(String serviceName, Collection<MethodDescriptor<
           "duplicate name %s", method.getFullMethodName());
     }
   }
+
+  /**
+   * Creates a new builder for a {@link ServiceDescriptor}.
+   *
+   * @since 1.1.0
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2666")
+  public static Builder newBuilder() {
```

Comment:
They both have pros and cons. I slightly lean towards not having name here, because it's more consistent. `build()` is usually very close to `newBuilder()`, so I think it should be easy to make sure `setName()` is called. `build()` should have a javadoc says that `setName()` must be called.